eclipse-che/che-server
# Respect authorisation request opt-out on workspace start

### What does this PR do?
If user rejects an scm provider authorisation request while creating or starting existed workspace store the name of the scm provider in the `workspace-preferences` config-map. The workspace create/start step must proceed without token fetch step. If user creates another workspace or starts existed workspace from an scm provider which name is stored in the config-map, do not ask the authorisation as it was already rejected once.

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/22499

### How to test this PR?

1. Deploy che from the PR image: `quay.io/eclipse/che-server:pr-576`
2. Setup [GitHub oauth flow](https://eclipse.dev/che/docs/stable/administration-guide/configuring-oauth-2-for-github/)
3. Start a workspace from a public GitHub repository url.
4. When the authorisation page opens, click the `cancel` button.
5. When workspace starts, stop it and then start it, see: workspace starts without asking the authorisation.
6. Delete the workspace and create it again, see: workspace starts without asking the authorisation.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
